pygmt/src/shift_origin.py

```diff
@@ -2,20 +2,42 @@
 shift_origin - Shift plot origin in x and/or y directions.
 """
 
+from contextlib import contextmanager
+
 from pygmt.clib import Session
+from pygmt.helpers import build_arg_list
 
 
 def shift_origin(
     self, xshift: float | str | None = None, yshift: float | str | None = None
 ):
     r"""
-    Shift plot origin in x and/or y directions.
+    Shift the plot origin in x and/or y directions.
+
+    The shifts can be temporary or permanent. If used as a context manager, the shifts
```

Comment:
Wonder if it is better to change the order to first mention permanent and then temporary, as in the examples at the end.
At least I would have this always in the same order.